### DISCUSSION TITLE
Gameplay rendering differs between stable and lazer

### DISCUSSION THREAD

**Discussion by dengr1065** [ppy/osu#23704] [Q&A] [answered]
The slider circle and combo number appear smooth in stable, but are slightly jagged and offset to the left in lazer. Most likely this is caused by lazer using high resolution (`@2x`) sprites regardless of screen resolution. The screen resolution I use is 1366x768, therefore stable is using low resolution sprites by default. I'm not sure whether skin is the root cause of this difference. If that's the case, please disregard this.

Possibly related: https://github.com/ppy/osu/discussions/20723

|Stable (low resolution)|Lazer|
|------|-----|
|![Stable](https://github.com/ppy/osu/assets/32245793/0c45c3b1-9ba6-4034-97f4-9b2c677fc5ba)|![Lazer](https://github.com/ppy/osu/assets/32245793/898a82d8-72ed-495f-9960-0008b78cb1ea)|

Skin in the screenshots: [BTMC: FREEDOM DiVE](https://github.com/BTMC-techteam/skins/releases/latest/download/-.BTMC.Freedom.Dive.osk)
Lazer version: 2023.513.0

**Comment by peppy** [comment, marked as answer]
The minor differences you are seeing may be due to the mipmap weighting we have applied. But I don't foresee this changing, and I don't think the difference is enough that we need to offer a setting for this.

On a per skin basis you can fix this yourself – just delete all `@2x` files and the game will only load the low resolution ones.

Here's an example for this skin:

[BTMC.Freedom.Drive (low resolution).osk.zip](https://github.com/ppy/osu/files/11611802/BTMC.Freedom.Drive.low.resolution.osk.zip)
